File: web/nextjs/lib/api/career.ts
import { apiClient } from "@/lib/api/client";
import type {
  CareerApplication,
  CareerArtifact,
  CareerBlacklistEntry,
  CareerDoctorReport,
  CareerEvaluateResult,
  CareerEvaluation,
  CareerFollowup,
  CareerIntakeProposal,
  CareerInterviewPrep,
  CareerOfferPrep,
  CareerPatterns,
  CareerPipelineItem,
  CareerPortal,
  CareerProfile,
  CareerSalaryGap,
  CareerStory,
  Paginated,
} from "@/types/career";

export async function getCareerProfile() {
  const { data } = await apiClient.get<CareerProfile>("/career/profile");
  return data;
}

export async function patchCareerProfile(payload: Record<string, unknown>) {
  const { data } = await apiClient.patch<CareerProfile>("/career/profile", payload);
  return data;
}

export async function careerIntake(document: string) {
  const { data } = await apiClient.post<CareerIntakeProposal>("/career/intake", {
    document,
  });
  return data;
}

export async function evaluateCareer(payload: {
  job_url?: string;
  jd_text?: string;
  mode?: string;
  tailor_cv?: boolean;
  confirm_blacklist?: boolean;
}) {
  const { data } = await apiClient.post<CareerEvaluateResult>("/career/evaluate", payload, {
    timeout: 180_000,
  });
  return data;
}

export async function listCareerEvaluations() {
  const { data } = await apiClient.get<Paginated<CareerEvaluation>>("/career/evaluations", {
    params: { per_page: 50 },
  });
  return data;
}

export async function getCareerEvaluation(id: string) {
  const { data } = await apiClient.get<CareerEvaluation>(`/career/evaluations/${id}`);
  return data;
}

export async function listCareerPipeline() {
  const { data } = await apiClient.get<Paginated<CareerPipelineItem>>("/career/pipeline", {
    params: { per_page: 50 },
  });
  return data;
}

export async function listCareerTracker(status?: string) {
  const { data } = await apiClient.get<Paginated<CareerApplication>>("/career/applications", {
    params: { per_page: 50, status: status || undefined },
  });
  return data;
}

export async function setCareerStatus(id: string, status: string, note?: string) {
  const { data } = await apiClient.post<CareerApplication>(
    `/career/applications/${id}/status`,
    { status, note }
  );
  return data;
}

export async function careerDoctor() {
  const { data } = await apiClient.get<CareerDoctorReport>("/career/doctor");
  return data;
}

export async function careerPatterns() {
  const { data } = await apiClient.get<CareerPatterns>("/career/patterns");
  return data;
}

export async function careerScan(payload: {
  board?: string;
  slug?: string;
  company?: string;
  query?: string;
}) {
  const { data } = await apiClient.post("/career/scan", payload, { timeout: 120_000 });
  return data;
}

export async function listCareerPortals() {
  const { data } = await apiClient.get<CareerPortal[]>("/career/portals");
  return data;
}

export async function addCareerPortal(payload: {
  board: string;
  slug: string;
  company?: string;
}) {
  const { data } = await apiClient.post<CareerPortal>("/career/portals", payload);
  return data;
}

export async function listCareerBlacklist() {
  const { data } = await apiClient.get<CareerBlacklistEntry[]>("/career/blacklist");
  return data;
}

export async function addCareerBlacklist(payload: {
  company?: string;
  domain?: string;
  reason?: string;
}) {
  const { data } = await apiClient.post<CareerBlacklistEntry>("/career/blacklist", payload);
  return data;
}

export async function tailorCareerCV(evaluationId: string) {
  const { data } = await apiClient.post<CareerArtifact>(`/career/evaluations/${evaluationId}/cv`, {}, {
    timeout: 120_000,
  });
  return data;
}

export async function careerCoverLetter(evaluationId: string) {
  const { data } = await apiClient.post<CareerArtifact>(
    `/career/evaluations/${evaluationId}/cover`,
    {},
    { timeout: 120_000 }
  );
  return data;
}

export async function careerEmailDraft(evaluationId: string) {
  const { data } = await apiClient.post<CareerArtifact>(
    `/career/evaluations/${evaluationId}/email`,
    {},
    { timeout: 120_000 }
  );
  return data;
}

export async function listCareerArtifacts(applicationId: string) {
  const { data } = await apiClient.get<CareerArtifact[]>(
    `/career/applications/${applicationId}/artifacts`
  );
  return data;
}

export async function listCareerStories() {
  const { data } = await apiClient.get<CareerStory[]>("/career/stories");
  return data;
}

export async function upsertCareerStory(payload: Partial<CareerStory> & { title: string }) {
  const { data } = await apiClient.post<CareerStory>("/career/stories", payload);
  return data;
}

export async function listCareerFollowups() {
  const { data } = await apiClient.get<CareerFollowup[]>("/career/followups");
  return data;
}

export async function careerFollowup(applicationId: string) {
  const { data } = await apiClient.post<CareerFollowup>(
    `/career/applications/${applicationId}/followup`
  );
  return data;
}

export async function careerInterviewPrep(applicationId: string) {
  const { data } = await apiClient.post<CareerInterviewPrep>(
    `/career/applications/${applicationId}/interview-prep`
  );
  return data;
}

export async function careerOfferPrep(applicationId: string) {
  const { data } = await apiClient.post<CareerOfferPrep>(
    `/career/applications/${applicationId}/offer-prep`
  );
  return data;
}

export async function careerSalaryGap(
  applicationId: string,
  advertised?: string,
  actual?: string
) {
  const { data } = await apiClient.post<CareerSalaryGap>(
    `/career/applications/${applicationId}/salary-gap`,
    { advertised, actual }
  );
  return data;
}

export async function careerBatchEvaluate() {
  const { data } = await apiClient.post("/career/pipeline/batch", { limit: 8 }, { timeout: 180_000 });
  return data;
}

File: server/internal/model/career.go
package model

import (
	"time"

	"github.com/google/uuid"
)

const AgentNameCareerOps = "CareerOps"

// Application tracker states. Transitions are enforced in package career.
const (
	CareerStatusEvaluated = "evaluated"
	CareerStatusApplied   = "applied"
	CareerStatusResponded = "responded"
	CareerStatusInterview = "interview"
	CareerStatusOffer     = "offer"
	CareerStatusRejected  = "rejected"
	CareerStatusDiscarded = "discarded"
	CareerStatusSkip      = "skip"
	CareerStatusHired     = "hired"
)

const (
	CareerPipelineOpen        = "open"
	CareerPipelineClosed      = "closed"
	CareerPipelineBlacklisted = "blacklisted"
)

const (
	CareerEvalModeFull   = "full"
	CareerEvalModeTriage = "triage"
)

const (
	CareerArtifactCV      = "cv"
	CareerArtifactCover   = "cover"
	CareerArtifactEmail   = "email"
	CareerArtifactAnswers = "answers"
)

const (
	CareerStoryCV      = "cv"
	CareerStoryUser    = "user_stated"
	CareerStoryDerived = "derived_unverified"
)

// RecommendApplyMin is the default floor for "apply". House rules may raise it,
// never lower it below this without an explicit profile override.
const RecommendApplyMin = 4.0

// RecommendFormAnswersMin is the floor for drafting Block H application answers.
const RecommendFormAnswersMin = 4.5

type CareerIdentity struct {
	FullName string   `json:"full_name,omitempty"`
	Email    string   `json:"email,omitempty"`
	Phone    string   `json:"phone,omitempty"`
	Links    []string `json:"links,omitempty"`
}

type CareerTargets struct {
	Titles     []string `json:"titles,omitempty"`
	Seniority  string   `json:"seniority,omitempty"`
	Industries []string `json:"industries,omitempty"`
	MinComp    string   `json:"min_comp,omitempty"`
}

type CareerLocation struct {
	Cities     []string `json:"cities,omitempty"`
	Remote     bool     `json:"remote,omitempty"`
	Relocation bool     `json:"relocation,omitempty"`
}

type CareerWorkAuth struct {
	Countries         []string `json:"countries,omitempty"`
	NeedsSponsorship  bool     `json:"needs_sponsorship,omitempty"`
	AuthorizedAlready bool     `json:"authorized_already,omitempty"`
}

type CareerProfile struct {
	ID          uuid.UUID      `json:"id"`
	OrgID       uuid.UUID      `json:"org_id"`
	UserID      uuid.UUID      `json:"user_id"`
	CVMarkdown  string         `json:"cv_markdown"`
	Identity    CareerIdentity `json:"identity"`
	Targets     CareerTargets  `json:"targets"`
	Location    CareerLocation `json:"location"`
	WorkAuth    CareerWorkAuth `json:"work_auth"`
	Voice       string         `json:"voice"`
	HouseRules  string         `json:"house_rules"`
	ProofPoints string         `json:"proof_points"`
	Narrative   string         `json:"narrative"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
}

type UpdateCareerProfileRequest struct {
	CVMarkdown  *string         `json:"cv_markdown,omitempty"`
	Identity    *CareerIdentity `json:"identity,omitempty"`
	Targets     *CareerTargets  `json:"targets,omitempty"`
	Location    *CareerLocation `json:"location,omitempty"`
	WorkAuth    *CareerWorkAuth `json:"work_auth,omitempty"`
	Voice       *string         `json:"voice,omitempty"`
	HouseRules  *string         `json:"house_rules,omitempty"`
	ProofPoints *string         `json:"proof_points,omitempty"`
	Narrative   *string         `json:"narrative,omitempty"`
}

type CareerPipelineItem struct {
	ID                uuid.UUID  `json:"id"`
	OrgID             uuid.UUID  `json:"org_id"`
	ProfileID         uuid.UUID  `json:"profile_id"`
	ListingURL        string     `json:"listing_url"`
	Company           string     `json:"company"`
	Title             string     `json:"title"`
	Source            string     `json:"source"`
	Status            string     `json:"status"`
	Liveness          string     `json:"liveness"`
	LivenessCheckedAt *time.Time `json:"liveness_checked_at,omitempty"`
	CreatedAt         time.Time  `json:"created_at"`
	UpdatedAt         time.Time  `json:"updated_at"`
}

type CareerApplication struct {
	ID         uuid.UUID `json:"id"`
	OrgID      uuid.UUID `json:"org_id"`
	ProfileID  uuid.UUID `json:"profile_id"`
	Company    string    `json:"company"`
	Role       string    `json:"role"`
	ListingURL string    `json:"listing_url"`
	Status     string    `json:"status"`
	Score      *float64  `json:"score,omitempty"`
	Via        string    `json:"via"`
	Agency     string    `json:"agency"`
	Employer   string    `json:"employer"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type CareerStatusEvent struct {
	ID            uuid.UUID `json:"id"`
	OrgID         uuid.UUID `json:"org_id"`
	ProfileID     uuid.UUID `json:"profile_id"`
	ApplicationID uuid.UUID `json:"application_id"`
	FromStatus    string    `json:"from_status"`
	ToStatus      string    `json:"to_status"`
	Note          string    `json:"note"`
	CreatedAt     time.Time `json:"created_at"`
}

// CareerScore is holistic 1–5 across five dimensions — not a formula of them.
// Block G (legitimacy) is stored on the evaluation, never mixed into Overall.
type CareerScore struct {
	Overall              float64            `json:"overall"`
	Dimensions           map[string]float64 `json:"dimensions,omitempty"`
	RecommendApply       bool               `json:"recommend_apply"`
	RecommendFormAnswers bool               `json:"recommend_form_answers"`
	Recommendation       string             `json:"recommendation,omitempty"`
}

type CareerEvalBlocks struct {
	A        string `json:"a,omitempty"`
	B        string `json:"b,omitempty"`
	C        string `json:"c,omitempty"`
	D        string `json:"d,omitempty"`
	E        string `json:"e,omitempty"`
	F        string `json:"f,omitempty"`
	G        string `json:"g,omitempty"`
	H        string `json:"h,omitempty"`
	WorkAuth string `json:"work_auth,omitempty"`
}

type CareerEvaluation struct {
	ID             uuid.UUID        `json:"id"`
	OrgID          uuid.UUID        `json:"org_id"`
	ProfileID      uuid.UUID        `json:"profile_id"`
	ApplicationID  *uuid.UUID       `json:"application_id,omitempty"`
	PipelineItemID *uuid.UUID       `json:"pipeline_item_id,omitempty"`
	ListingURL     string           `json:"listing_url"`
	JDText         string           `json:"jd_text"`
	Company        string           `json:"company"`
	Role           string           `json:"role"`
	Blocks         CareerEvalBlocks `json:"blocks"`
	Score          CareerScore      `json:"score"`
	ReportMarkdown string           `json:"report_markdown"`
	LegitimacyTier string           `json:"legitimacy_tier"`
	HardStop       bool             `json:"hard_stop"`
	HardStopReason string           `json:"hard_stop_reason"`
	Mode           string           `json:"mode"`
	CreatedAt      time.Time        `json:"created_at"`
}

type CareerArtifact struct {
	ID            uuid.UUID  `json:"id"`
	OrgID         uuid.UUID  `json:"org_id"`
	ProfileID     uuid.UUID  `json:"profile_id"`
	ApplicationID *uuid.UUID `json:"application_id,omitempty"`
	EvaluationID  *uuid.UUID `json:"evaluation_id,omitempty"`
	Kind          string     `json:"kind"`
	Title         string     `json:"title"`
	BodyMarkdown  string     `json:"body_markdown"`
	FileID        string     `json:"file_id,omitempty"`
	HasPDF        bool       `json:"has_pdf"`
	CreatedAt     time.Time  `json:"created_at"`
}

type CareerBlacklistEntry struct {
	ID        uuid.UUID `json:"id"`
	OrgID     uuid.UUID `json:"org_id"`
	ProfileID uuid.UUID `json:"profile_id"`
	Company   string    `json:"company"`
	Domain    string    `json:"domain"`
	Reason    string    `json:"reason"`
	CreatedAt time.Time `json:"created_at"`
}

type CareerPortal struct {
	ID           uuid.UUID `json:"id"`
	OrgID        uuid.UUID `json:"org_id"`
	ProfileID    uuid.UUID `json:"profile_id"`
	Board        string    `json:"board"`
	Slug         string    `json:"slug"`
	Company      string    `json:"company"`
	TitleInclude []string  `json:"title_include"`
	TitleExclude []string  `json:"title_exclude"`
	Enabled      bool      `json:"enabled"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

type CareerStory struct {
	ID         uuid.UUID `json:"id"`
	OrgID      uuid.UUID `json:"org_id"`
	ProfileID  uuid.UUID `json:"profile_id"`
	Title      string    `json:"title"`
	Situation  string    `json:"situation"`
	Task       string    `json:"task"`
	Action     string    `json:"action"`
	Result     string    `json:"result"`
	Reflection string    `json:"reflection"`
	Provenance string    `json:"provenance"`
	Tags       []string  `json:"tags"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type CareerScanRun struct {
	ID           uuid.UUID  `json:"id"`
	OrgID        uuid.UUID  `json:"org_id"`
	ProfileID    uuid.UUID  `json:"profile_id"`
	Status       string     `json:"status"`
	Added        int        `json:"added"`
	Skipped      int        `json:"skipped"`
	ErrorMessage string     `json:"error_message,omitempty"`
	CreatedAt    time.Time  `json:"created_at"`
	CompletedAt  *time.Time `json:"completed_at,omitempty"`
}

type CareerDoctorReport struct {
	OK       bool     `json:"ok"`
	Warnings []string `json:"warnings"`
	Info     []string `json:"info"`
}

type CareerIntakeProposal struct {
	Summary string                     `json:"summary"`
	Patch   UpdateCareerProfileRequest `json:"patch"`
}

type EvaluateCareerRequest struct {
	JobURL           string `json:"job_url,omitempty"`
	JDText           string `json:"jd_text,omitempty"`
	Mode             string `json:"mode,omitempty"` // full | triage
	TailorCV         bool   `json:"tailor_cv,omitempty"`
	ConfirmBlacklist bool   `json:"confirm_blacklist,omitempty"`
}

type CareerEvaluateResult struct {
	Evaluation   *CareerEvaluation     `json:"evaluation"`
	Application  *CareerApplication    `json:"application,omitempty"`
	BlacklistHit *CareerBlacklistEntry `json:"blacklist_hit,omitempty"`
	Dead         bool                  `json:"dead,omitempty"`
	DeadReason   string                `json:"dead_reason,omitempty"`
	Artifacts    []CareerArtifact      `json:"artifacts,omitempty"`
}

type ScanCareerRequest struct {
	Board   string `json:"board,omitempty"` // greenhouse | ashby | lever
	Slug    string `json:"slug,omitempty"`
	Company string `json:"company,omitempty"`
	Query   string `json:"query,omitempty"` // extra title include
}

type CareerScanResult struct {
	Run   *CareerScanRun       `json:"run"`
	Added []CareerPipelineItem `json:"added"`
}

type SetCareerStatusRequest struct {
	Status string `json:"status" validate:"required"`
	Note   string `json:"note,omitempty"`
}

type AddCareerBlacklistRequest struct {
	Company string `json:"company,omitempty"`
	Domain  string `json:"domain,omitempty"`
	Reason  string `json:"reason,omitempty"`
}

type AddCareerPortalRequest struct {
	Board        string   `json:"board"`
	Slug         string   `json:"slug"`
	Company      string   `json:"company,omitempty"`
	TitleInclude []string `json:"title_include,omitempty"`
	TitleExclude []string `json:"title_exclude,omitempty"`
}

type CareerContact struct {
	ID            uuid.UUID  `json:"id"`
	OrgID         uuid.UUID  `json:"org_id"`
	ProfileID     uuid.UUID  `json:"profile_id"`
	ApplicationID *uuid.UUID `json:"application_id,omitempty"`
	Name          string     `json:"name"`
	Role          string     `json:"role"`
	Company       string     `json:"company"`
	// Email is third-party PII. Never log it.
	Email         string    `json:"email,omitempty"`
	LinkedInURL   string    `json:"linkedin_url,omitempty"`
	Note          string    `json:"note,omitempty"`
	LinkedInDraft string    `json:"linkedin_draft,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
}

type AddCareerContactRequest struct {
	ApplicationID *uuid.UUID `json:"application_id,omitempty"`
	Name          string     `json:"name"`
	Role          string     `json:"role,omitempty"`
	Company       string     `json:"company,omitempty"`
	Email         string     `json:"email,omitempty"`
	LinkedInURL   string     `json:"linkedin_url,omitempty"`
	Note          string     `json:"note,omitempty"`
}

type CareerFollowup struct {
	ID            uuid.UUID `json:"id"`
	OrgID         uuid.UUID `json:"org_id"`
	ProfileID     uuid.UUID `json:"profile_id"`
	ApplicationID uuid.UUID `json:"application_id"`
	DueAt         time.Time `json:"due_at"`
	Kind          string    `json:"kind"`
	Draft         string    `json:"draft"`
	Sent          bool      `json:"sent"`
	CreatedAt     time.Time `json:"created_at"`
}

type CareerPatterns struct {
	Applications int            `json:"applications"`
	ByStatus     map[string]int `json:"by_status"`
	AvgScore     float64        `json:"avg_score"`
	SkillGaps    []string       `json:"skill_gaps,omitempty"`
}

type CareerInterviewPrep struct {
	Company        string        `json:"company"`
	Role           string        `json:"role"`
	Status         string        `json:"status,omitempty"`
	ScoreFloorMet  bool          `json:"score_floor_met"`
	Stories        []CareerStory `json:"stories"`
	PrepMarkdown   string        `json:"prep_markdown"`
	NeverSubmit    bool          `json:"never_submit"`
	NotLegalAdvice bool          `json:"not_legal_advice"`
}

type CareerOfferPrep struct {
	Company        string `json:"company"`
	Role           string `json:"role"`
	PrepMarkdown   string `json:"prep_markdown"`
	NotLegalAdvice bool   `json:"not_legal_advice"`
}

type CareerSalaryGap struct {
	Desired        string `json:"desired"`
	Advertised     string `json:"advertised"`
	Actual         string `json:"actual"`
	Note           string `json:"note"`
	NotLegalAdvice bool   `json:"not_legal_advice"`
}

type CareerSalaryObservation struct {
	ID            uuid.UUID  `json:"id"`
	OrgID         uuid.UUID  `json:"org_id"`
	ProfileID     uuid.UUID  `json:"profile_id"`
	ApplicationID *uuid.UUID `json:"application_id,omitempty"`
	Desired       string     `json:"desired"`
	Advertised    string     `json:"advertised"`
	Actual        string     `json:"actual"`
	CreatedAt     time.Time  `json:"created_at"`
}

type CareerOffer struct {
	ID            uuid.UUID      `json:"id"`
	OrgID         uuid.UUID      `json:"org_id"`
	ProfileID     uuid.UUID      `json:"profile_id"`
	ApplicationID uuid.UUID      `json:"application_id"`
	Clauses       map[string]any `json:"clauses,omitempty"`
	Notes         string         `json:"notes"`
	CreatedAt     time.Time      `json:"created_at"`
}

type CareerBatchResult struct {
	Evaluated int                    `json:"evaluated"`
	Skipped   int                    `json:"skipped"`
	Results   []CareerEvaluateResult `json:"results"`
}
